app/src/main/java/com/tip/lunchbox/data/service/APIService.java

```diff
@@ -0,0 +1,32 @@
+package com.tip.lunchbox.data.service;
+
+import com.tip.lunchbox.model.DailyMenuResponse;
+import com.tip.lunchbox.model.Restaurant;
+import com.tip.lunchbox.model.Reviews;
+import com.tip.lunchbox.model.SearchResponse;
+
+import java.util.List;
+import java.util.Map;
+
+import io.reactivex.rxjava3.core.Observable;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+import retrofit2.http.QueryMap;
+
+public interface APIService {
+
+    @GET("search")
+    Observable<SearchResponse> getSearchResponse(@QueryMap Map<String, String> searchParameters);
```

Comment:
@epicadk can you please explain the logic of passing a query map? From the documentation, I can see that search can be used be in multiple ways, should separate methods be defined for each?

Reply:
I thought we could do that in the repository class if required

Reply:
Should that be in the repository or the api interface itself?

Reply:
We can create a separate interface with all the different searching use cases inside it as different methods? What do you think?

Reply:
Yes but all search cases may not be used I suggest we leave it at this and if we do use a few of them more than once we create a separate Interface